modules/client/src/internal/utils.ts

```diff
@@ -824,3 +863,763 @@ export function toPluginPermissionOperationType(
+      // so we decode the initialize from function
+      const initializeFromDecodedParams = decodeInitializeFromAction(
+        upgradeToAndCallDecodedParams.data,
+      );
+      // check that the implementation address is the same as specified
+      // in the upgradeToAndCall action
+      if (
+        implementationAddress !==
+          upgradeToAndCallDecodedParams.implementationAddress
+      ) {
+        causes.push(
+          DaoUpdateProposalInvalidityCause
+            .INVALID_UPGRADE_TO_AND_CALL_IMPLEMENTATION_ADDRESS,
+        );
+      }
+      // check that the current version version of the dao is the same
+      // as the one specified as previous version in the initializeFrom function
+      if (
+        JSON.stringify(initializeFromDecodedParams.previousVersion) !==
+          JSON.stringify(currentDaoVersion)
+      ) {
+        causes.push(
+          DaoUpdateProposalInvalidityCause.INVALID_UPGRADE_TO_AND_CALL_VERSION,
+        );
+      }
+      // TODO
+      // check that the data can be decoded with the abi in the metadata of the version we are
+      // upgrading to. For now is not possible so we check that the data is empty
+      if (initializeFromDecodedParams.initData.length !== 0) {
+        causes.push(
+          DaoUpdateProposalInvalidityCause.INVALID_UPGRADE_TO_AND_CALL_DATA,
+        );
+      }
+    }
+  } else {
+    // add invalid actions to the causes array
+    causes.push(
+      DaoUpdateProposalInvalidityCause.INVALID_ACTIONS,
+    );
+  }
+  // return the validity of the proposal
+  return { isValid: causes.length === 0, causes };
+}
+
+export async function validateUpdatePluginProposalActions(
+  actions: DaoAction[],
+  daoAddress: string,
+  pspAddress: string,
+  graphql: IClientGraphQLCore,
+  ipfs: IClientIpfsCore,
+) {
+  let causes: PluginUpdateProposalInValidityCause[][] = [];
+  const classifiedActions = classifyProposalActions(actions);
+  if (isPluginUpdateAction(classifiedActions)) {
+    causes[0] = [];
+    for (const [index, action] of classifiedActions.entries()) {
+      switch (action) {
+        case ProposalActionTypes.GRANT_PLUGIN_UPDATE_PERMISSION:
+          causes[0].concat(
+            await validateGrantUpdatePluginPermissionAction(
+              actions[index],
+              pspAddress,
+              daoAddress,
+              graphql,
+            ),
+          );
+          break;
+        case ProposalActionTypes.REVOKE_PLUGIN_UPGRADE_PERMISSION:
+          causes[0].concat(
+            await validateRevokeUpdatePluginPermissionAction(
+              actions[index],
+              pspAddress,
+              daoAddress,
+              graphql,
+            ),
+          );
+          break;
+        case ProposalActionTypes.APPLY_UPDATE:
+          const resCauses = await validateApplyUpdateFunction(
+            actions[index],
+            daoAddress,
+            graphql,
+            ipfs,
+          );
+          causes[0].concat(
```

Comment:
concat doesn't update the first array but only returns the merged array
```suggestion
          causes[0] = causes[0].concat(
```